native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java

```diff
@@ -84,14 +82,9 @@ public static BStream nativeQuery(Environment env, BObject client, BObject param
         boolean withinTrxBlock = Utils.isWithinTrxBlock(trxResourceManager);
         boolean trxManagerEnabled = trxResourceManager.getTransactionManagerEnabled();
         TransactionLocalContext currentTrxContext = trxResourceManager.getCurrentTransactionContext();
-        Future balFuture = env.markAsync();
-        SQL_EXECUTOR_SERVICE.execute(() -> {
```

Comment:
shall we remove SQL_EXECUTOR_SERVICE too

Reply:
Fixed with 25ea79e5bc9e354f4769f5b3ec71c0f3270da2e6